python/lsst/ip/diffim/subtractImages.py

```diff
@@ -167,7 +185,42 @@ def __init__(self, **kwargs):
         self.convolutionControl.setDoNormalize(False)
         self.convolutionControl.setDoCopyEdge(True)
 
-    def run(self, template, science, sources):
+    def _applyExternalCalibrations(self, exposure, finalizedPsfApCorrCatalog):
+        """Replace calibrations (psf, and ApCorrMap) on this exposure with external ones.".
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Input exposure to adjust calibrations.
+        finalizedPsfApCorrCatalog : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with finalized psf models and aperture correction
+            maps to be applied if config.doApplyFinalizedPsf=True.  Catalog uses
+            the detector id for the catalog id, sorted on id for fast lookup.
+
+        Returns
+        -------
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Exposure with adjusted calibrations.
+        """
+        detectorId = exposure.info.getDetector().getId()
+
+        row = finalizedPsfApCorrCatalog.find(detectorId)
+        if row is None:
+            self.log.warning("Detector id %s not found in finalizedPsfApCorrCatalog; "
+                             "Using original psf.", detectorId)
+        else:
+            psf = row.getPsf()
+            apCorrMap = row.getApCorrMap()
+            if psf is None or apCorrMap is None:
+                self.log.warning("Detector id %s has None for psf/apCorrMap in "
```

Comment:
I'd rather this log message be more explicit about which of those are None, to help with debugging. Either two separate `if` statements, or some logic in the log message, something like '"psf=None" if psf is None else ""`.